xcode/Subconscious/Shared/Components/AddressBook/DidQrCodeView.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  DidQrCodeView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 27/2/2023.
+//
+
+import Foundation
+import SwiftUI
+import CoreImage.CIFilterBuiltins
+
+struct DidQrCodeView: View {
+    var did: Did
+    
+    let context = CIContext()
+    let filter = CIFilter.qrCodeGenerator()
+    let colorInvertFilter = CIFilter.colorInvert()
+    let maskToAlphaFilter = CIFilter.maskToAlpha()
+    let colorReplaceFilter = CIFilter.spotColor()
+    
+    let maskFilter = CIFilter.blendWithMask()
+    
+    func generateQRCode(from string: String) -> UIImage? {
+        filter.message = Data(string.utf8)
+        
+        colorInvertFilter.inputImage = filter.outputImage
+        guard let inverted = colorInvertFilter.outputImage else { return nil }
```

Comment:
Invert to white-on-black to use it as a blending mask (where black is full transparency and white full opacity)